Versioning.md

```diff
@@ -0,0 +1,38 @@
+# iTwin.js versioning policies
+
+iTwin.js employs versioning policies to faciliate the following objectives:
+
+- Provide a broad set of stable, well-supported APIs.
+- Continuously deliver new features, bug fixes, and security updates.
+- Evolve APIs over time in response to user feedback.
+- Encourage users to keep up to date with the most recent versions.
+
+## API promotion process
+
+Each API exported by a package is marked with a [release tag](./docs/learning/guidelines/release-tags-guidelines.md) indicating its level of stability. A package's "public" API comprises all of its exported APIs that have been marked with the `@public` release tag. All other APIs are considered "non-public". The [package versioning](#package-versioning-policy) and [API deprecation](#api-deprecation-policy) policies apply only the a package's public API.
```

Comment:
What's the classification you're giving "non-public" here? I feel that just confuses things when you talk about `internal` below.

Reply:
"non-public" = not "@public".
I will clean this up to describe each tag in a consumer-facing manner.
I'd prefer not to talk about internal and alpha at all here, since they are omitted from the docs. But they're going to show up in auto-complete in IDEs so probably need to be defined.

Reply:
when you did the investigation into the APIs, how many were `alpha`?

I feel that it isn't used properly (i.e an excuse not to write docs) and based on your description below could really just be removed and instead we only use `internal` `beta` and `public` as the main release tags.

> "non-public" = not "@public".

I don't think that clarification is necessary and that sentence doesn't add much so could be removed.

Reply:
There are 774 occurrences of `@alpha` in common/api. I would like to see that number reduced significantly as we reevaluate our current API tags, but that effort is going to be much bigger than the scope of this PR.

I don't recall why extensions use `@preview` instead of `@beta` - do you? Was there some case in which some API would be `@preview` but not also `@beta`?

Reply:
> I don't recall why extensions use @preview instead of @beta - do you? Was there some case in which some API would be @preview but not also @beta?

We separated them because we wanted to leave the potential for a `@public` API to be added to the Extension API as `@preview` in case we weren't sure we wanted to keep it in there. However, we have nothing at all in preview today and I'm re-thinking my stance on that now... Feels like it's an unnecessary complication to add onto an already confusing distinction between the two APIs. 

I still think we need a "preview" Extension API but maybe that should be for things tagged `@beta`.